src/snappy.c

```diff
@@ -126,7 +126,55 @@ static inline bool is_little_endian(void)
 	return false;
 }
 
-#ifndef _MSC_VER
+#if defined(_AIX)
```

Comment:
This should *probably* be done in configure but we might get away with it here.
I think there's also an important point to separate toolchain from platform, for example the gcc toolchain will most likely provide proper builtins on all platforms.